File: features/merchant/favorite/src/main/java/com/tokopedia/favorite/di/modul/UserSessionModule.java
package com.tokopedia.favorite.di.modul;

import android.content.Context;

import com.tokopedia.abstraction.common.di.qualifier.ApplicationContext;
import com.tokopedia.favorite.di.scope.FavoriteScope;
import com.tokopedia.user.session.UserSession;
import com.tokopedia.user.session.UserSessionInterface;

import dagger.Module;
import dagger.Provides;

@FavoriteScope
@Module
public class UserSessionModule {

    @FavoriteScope
    @Provides
    UserSessionInterface provideUserSession(@ApplicationContext Context context) {
        return new UserSession(context);
    }
}

File: features/tokopedianow/src/main/java/com/tokopedia/tokopedianow/searchcategory/di/UserSessionModule.kt
package com.tokopedia.tokopedianow.searchcategory.di

import android.content.Context
import com.tokopedia.user.session.UserSession
import com.tokopedia.user.session.UserSessionInterface
import dagger.Module
import dagger.Provides

@Module
class UserSessionModule {

    @Provides
    fun provideUserSession(context: Context): UserSessionInterface {
        return UserSession(context)
    }
}